File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.go
func maxNumberOfBalloons(text string) int {
    freq := make([]int, 26)
    
    for _, c := range text {
        freq[c-'a']++
    }
    
    b := freq['b'-'a']
    a := freq['a'-'a']
    l := freq['l'-'a'] / 2
    o := freq['o'-'a'] / 2
    n := freq['n'-'a']
    
    return min(b, a, l, o, n)
}

func min(values ...int) int {
    minVal := values[0]
    for _, v := range values[1:] {
        if v < minVal {
            minVal = v
        }
    }
    return minVal
}

File: 1189-maximum-number-of-balloons/1189-maximum-number-of-balloons.js
/**
 * @param {string} text
 * @return {number}
 */
var maxNumberOfBalloons = function(text) {
    const count = {};
    
    for (let char of text) {
        count[char] = (count[char] || 0) + 1;
    }

    const b = count['b'] || 0;
    const a = count['a'] || 0;
    const l = Math.floor((count['l'] || 0 ) / 2);
    const o = Math.floor((count['o'] || 0) / 2);
    const n = count['n'] || 0;

    return Math.min(b, a, l, o, n);
};